src/serde_impl.rs

```diff
@@ -0,0 +1,227 @@
+use ff::PrimeField;
+use group::GroupEncoding;
+use serde::{de::Error as DeserializeError, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{
+    curves::{EpAffine, EqAffine},
+    fields::{Fp, Fq},
+};
+
+const ERR_CODE: &str = "deserialized bytes don't encode a field element";
+
+impl Serialize for Fp {
+    fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        self.to_repr().serialize(s)
+    }
+}
+
+impl<'de> Deserialize<'de> for Fp {
+    fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        let bytes = <[u8; 32]>::deserialize(d)?;
+        match Fp::from_repr(bytes).into() {
+            Some(fp) => Ok(fp),
+            None => Err(D::Error::custom(ERR_CODE)),
+        }
+    }
+}
+
+impl Serialize for Fq {
+    fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        self.to_repr().serialize(s)
+    }
+}
+
+impl<'de> Deserialize<'de> for Fq {
+    fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        let bytes = <[u8; 32]>::deserialize(d)?;
+        match Fq::from_repr(bytes).into() {
+            Some(fq) => Ok(fq),
+            None => Err(D::Error::custom(ERR_CODE)),
+        }
+    }
+}
+
+impl Serialize for EpAffine {
+    fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        self.to_bytes().serialize(s)
+    }
+}
+
+impl<'de> Deserialize<'de> for EpAffine {
+    fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        let bytes = <[u8; 32]>::deserialize(d)?;
+        match EpAffine::from_bytes_unchecked(&bytes).into() {
```

Comment:
```suggestion
        match EpAffine::from_bytes(&bytes).into() {
```
If you look at the implementation in https://github.com/zcash/pasta_curves/blob/main/src/curves.rs , `from_bytes_unchecked` and `from_bytes` have exactly the same behaviour, but using `from_bytes` is clearer because in fact the representation is checked.